<h1>BACHELOR OF SCIENCE IN INFORMATION TECHNOLOGY</h1>
<hr>
<h3>Pajarillon, Maria Louizza B.</h3>

In [1]:
import numpy as np
import pandas as pd

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv') 

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

In [6]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

<h1>Cleaning the Data</h1>

In [7]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1>Selecting a column</h1>
<h2>Syntax</h2>
data_frame = [' ']

In [8]:
penguins['island']

0      Torgersen
1      Torgersen
2      Torgersen
3      Torgersen
4      Torgersen
         ...    
339        Dream
340        Dream
341        Dream
342        Dream
343        Dream
Name: island, Length: 344, dtype: object

In [9]:
[penguins['island'] == 'Dream']

[0      False
 1      False
 2      False
 3      False
 4      False
        ...  
 339     True
 340     True
 341     True
 342     True
 343     True
 Name: island, Length: 344, dtype: bool]

<h1>Selecting Columns</h1>

In [10]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [11]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']

In [12]:
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [13]:
adelie_df = penguins[penguins['species'] == 'Adelie']

In [14]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [15]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']

In [16]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [17]:
penguins [(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [18]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [19]:
penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

0      39.10000
1      39.50000
2      40.30000
3      43.92193
4      36.70000
         ...   
339    55.80000
340    43.50000
341    49.60000
342    50.80000
343    50.20000
Name: bill_length_mm, Length: 344, dtype: float64

In [20]:
penguins['sex'] = penguins ['sex'].fillna(penguins['sex'].mode()[0])

In [21]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       2
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
dtype: int64

In [22]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [23]:
penguins ['species'] = penguins ['species'].astype('category')
penguins ['year'] = penguins ['year'].astype('category')


In [24]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     342 non-null    float64 
 3   bill_depth_mm      342 non-null    float64 
 4   flipper_length_mm  342 non-null    float64 
 5   body_mass_g        342 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    category
dtypes: category(2), float64(4), object(2)
memory usage: 17.2+ KB


In [25]:
penguins ['mass_kg'] = np.round(penguins['body_mass_g'] / 1000,1)

In [26]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  mass_kg  
0         3750.0    male  2007      3.8  
1         3800.0  female  2007      3.8  
2         3250.0  female  2007      3.2  
3            NaN    male  2007      NaN  
4         3450.0  female  2007      3.4  
..           ...     ...   ...      ...  
339       4000.0    male  2009      4.0  
340       3400.0  female  2009      3.4  
341       3775.0    male  2009      3.8  
342       4100.0    male  2009      4.1  
343       3775.0  female  2009      3.8  

[344 rows x 9 columns]

<h1>EXPLORATORY DATA ANALYTICS</h1>

<h3>1. Which species has the heaviest body mass in kg?</h3>

In [27]:
chinstrap_df['body_mass_g'].mean()

np.float64(3733.0882352941176)

In [28]:
adelie_df['body_mass_g'].mean()

np.float64(3700.662251655629)

In [29]:
gentoo_df['body_mass_g'].mean()

np.float64(5076.016260162602)

<h3>INSIGHT: Among the three species, Gentoo penguins have the heaviest average body mass at around 5.08 kg, compared to Chinstrap (3.73 kg) and Adelie (3.70 kg). This indicates that Gentoos are generally the largest species in terms of body weight. </h3>

<h3>2. Which island has the most penguins?</h3>

In [30]:
island_count = penguins['island'].value_counts()

In [31]:
island_count

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<H3>INSIGHT: The data shows that Biscoe Island hosts the largest number of penguins (168), making it the most populated island compared to Dream and Torgersen.>

<h3> 3. How many missing values are in each column? </h3>

In [32]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       2
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          2
sex                  0
year                 0
mass_kg              2
dtype: int64

<h3>4. How many male and female penguins are there?</h3>

In [33]:
penguins['sex'].value_counts()

sex
male      179
female    165
Name: count, dtype: int64

<h3> INSIGHT: Although the dataset is mostly complete, the sex column has the highest number of missing values (11), which means information about some penguins’ gender is not available.</h3>

<h3>5. What years are included in the dataset?</h3>

In [34]:
penguins['year'].unique()

[2007, 2008, 2009]
Categories (3, int64): [2007, 2008, 2009]

<H3> INSIGHT: he dataset includes penguin records from the years 2007, 2008, and 2009.</H3>

<h3>6.How many penguins were recorded in each year?</h3>

In [35]:
penguins['year'].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h3> INSIGHT: Among these years, the highest number of penguins was recorded in 2009 (120), followed by 2008 (114) and 2007 (110).</h3>

<h3>7. What is the most common island?</h3>

In [37]:
penguins['island'].mode()

0    Biscoe
Name: island, dtype: object

<h3>INSIGHT: The most common island in the dataset is Biscoe, which appears most frequently compared to the other islands.</h3>

<H1> DATA AGGREGATION</H1>
<HR></HR>
<H1> .groupby()</H1>

<h1>Q8. What is the average bill depth per species?</h1>

In [38]:
penguins.groupby("species")["bill_depth_mm"].mean()


C:\Users\Admin\AppData\Local\Temp\ipykernel_10928\3406580581.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins.groupby("species")["bill_depth_mm"].mean()


species
Adelie       18.346358
Chinstrap    18.420588
Gentoo       14.982114
Name: bill_depth_mm, dtype: float64

<h1>Insight: Based from the result, Chinstrap penguins have deeper bills on average compared to Gentoo and Adelie.</h1>

<h1>Q9. What is the maximum body mass per island?</h1>

In [39]:
penguins.groupby("island")["body_mass_g"].max()


island
Biscoe       6300.0
Dream        4800.0
Torgersen    4700.0
Name: body_mass_g, dtype: float64

<h1>Insight: All three years (2007–2009) contain penguins from multiple species, showing consistent data collection. The maximum body mass for Biscoe is 6300.0, 4800.0 for Dream, and 4700.0 for Torgersen.</h1>

<H1> DATA AGGREGATION</H1>
<HR></HR>
<H1> .corr()</H1>

<h1>Q10. Is there a correlation between bill depth and bill length?</h1>

In [40]:
bill_length_mm_corr = penguins['bill_depth_mm'].corr(penguins['bill_length_mm'])


In [41]:
bill_length_mm_corr

np.float64(-0.2350528703555326)

<h1>INSIGHT: The result shows a weak negative correlation, meaning penguins with deeper bills tend to have slightly shorter bill lengths.</h1>

<H1> DATA AGGREGATION</H1>
<HR></HR>
<H1> One Way Anova</H1>

In [42]:
from scipy.stats import f_oneway


<h1>Q11. Is there a significant difference in the bill length between different penguin species? </h1>

In [43]:
penguin_anova = penguins.dropna(subset=['bill_length_mm', 'species'])

In [44]:
adelie = penguin_anova[penguin_anova['species'] == 'Adelie']['bill_length_mm']

In [45]:
chinstrap = penguin_anova[penguin_anova['species'] == 'Chinstrap']['bill_length_mm']

In [46]:
gentoo = penguin_anova[penguin_anova['species'] == 'Gentoo']['bill_length_mm']

In [47]:
result = f_oneway(adelie, chinstrap, gentoo)

In [48]:
result

F_onewayResult(statistic=np.float64(410.6002550405077), pvalue=np.float64(2.6946137388895484e-91))

<h1>INSIGHT: It shows a significant difference in bill length among penguin species (very small p-value), meaning that species have distinct bill sizes rather than the differences occurring by chance.</h1>

<h1>Q12.Is there a significant difference in flipper length (flipper_length_mm) between male and female penguins?</h1>

In [50]:
anova2 = penguins.dropna(subset=['flipper_length_mm', 'sex'])


In [51]:
male = anova2[anova2['sex'] == 'male']['flipper_length_mm']
female = anova2[anova2['sex'] == 'female']['flipper_length_mm']

In [55]:
flipperlength_and_sex = f_oneway(male, female)


In [56]:
flipperlength_and_sex

F_onewayResult(statistic=np.float64(21.564111321582978), pvalue=np.float64(4.8966052232549966e-06))

<h1>INSIGHT: It shows a correlation between sex and flipper length, indicating that male and female penguins have significantly different flipper lengths.</h1>